connector/ldap/ldap.go

```diff
@@ -2,20 +2,109 @@
 package ldap
 
 import (
+	"encoding/json"
 	"errors"
 	"fmt"
+	"log"
 
 	"gopkg.in/ldap.v2"
 
 	"github.com/coreos/dex/connector"
 )
 
-// Config holds the configuration parameters for the LDAP connector.
+// Config holds the configuration parameters for the LDAP connector. The LDAP
+// connectors require executing two queries, the first to find the user based on
+// the username and password given to the connector. The second to use the user
+// entry to search for groups.
+//
+// An example config:
+//
+//     type: ldap
+//     config:
+//       host: ldap.example.com:636
+//       rootCA: /etc/dex/ldap.ca
+//       bindDN: uid=seviceaccount,cn=users,dc=example,dc=com
+//       bindPW: password
+//       userSearch:
+//         # Would translate to the query "(&(objectClass=person)&(uid=<username>))"
+//         baseDN: cn=users,dc=example,dc=com
+//         filter: "(objectClass=person)"
+//         username: uid
```

Comment:
I'm not sure how much of this is just ldap or how much is your adaptation. I think of username and uid as different